src/libraries/System.Private.CoreLib/src/System/Reflection/InvokerEmitUtil.cs

```diff
@@ -0,0 +1,201 @@
+            else
+            {
+                RuntimeType returnType;
+                if (method is RuntimeMethodInfo rmi)
+                {
+                    returnType = (RuntimeType)rmi.ReturnType;
+                }
+                else
+                {
+                    Debug.Assert(method is DynamicMethod);
+                    returnType = (RuntimeType)((DynamicMethod)method).ReturnType;
+                }
+
+                if (returnType == typeof(void))
+                {
+                    il.Emit(OpCodes.Ldnull);
+                }
+                else if (returnType.IsValueType)
+                {
+                    il.Emit(OpCodes.Box, returnType);
+                }
+                else if (returnType.IsPointer)
+                {
+                    il.Emit(OpCodes.Ldtoken, returnType);
+                    il.Emit(OpCodes.Call, Methods.Type_GetTypeFromHandle());
+                    il.Emit(OpCodes.Call, Methods.Pointer_Box());
+                }
+                else if (returnType.IsByRef)
+                {
+                    // Check for null ref return.
+                    Type elementType = returnType.GetElementType()!;
+                    Label retValueOk = il.DefineLabel();
+                    il.Emit(OpCodes.Dup);
+                    il.Emit(OpCodes.Brtrue_S, retValueOk);
+                    il.Emit(OpCodes.Call, Methods.ThrowHelper_Throw_NullReference_InvokeNullRefReturned());
+                    il.MarkLabel(retValueOk);
+
+                    // Handle per-type differences.
+                    if (elementType.IsValueType)
+                    {
+                        il.Emit(OpCodes.Ldobj, elementType);
+                        il.Emit(OpCodes.Box, elementType);
+                    }
+                    else if (elementType.IsPointer)
+                    {
+                        il.Emit(OpCodes.Ldind_Ref);
+                        il.Emit(OpCodes.Conv_U);
+                        il.Emit(OpCodes.Ldtoken, elementType);
+                        il.Emit(OpCodes.Call, Methods.Type_GetTypeFromHandle());
+                        il.Emit(OpCodes.Call, Methods.Pointer_Box());
+                    }
+                    else
+                    {
+                        il.Emit(OpCodes.Ldobj, elementType);
+                    }
+                }
+            }
+
+            il.Emit(OpCodes.Ret);
+
+            // Create the delegate; it is also compiled at this point due to restrictedSkipVisibility=true.
+            return (InvokeFunc)dm.CreateDelegate(typeof(InvokeFunc), target: null);
+        }
+
+        private static RuntimeType NormalizePointerType(RuntimeType type)
+        {
+            if (type.IsPointer)
+            {
+                type = (RuntimeType)typeof(IntPtr);
+            }
+            else if (type.IsByRef && RuntimeTypeHandle.GetElementType(type).IsPointer)
+            {
+                type = (RuntimeType)typeof(IntPtr).MakeByRefType();
+            }
```

Comment:
```suggestion
```
I do not see why this should be needed.

Reply:
Once you delete this, this will be a trivial method that is only called from one place so it can be just inlined in that one place.

Reply:
Thanks. Verified locally by running tests using `#define USE_EMIT_INVOKE` in both the method and ctor invokers.